test/lib/abbot/testFreeStyleLibreData.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+/* global beforeEach, describe, it */
+
+import {expect} from 'salinity';
+
+import builder from '../../../lib/objectBuilder';
+import annotate from '../../../lib/eventAnnotations';
+
+import {FreeStyleLibreData} from '../../../lib/drivers/abbott/freeStyleLibreData';
+import {
+  GLUCOSE_HI,
+  GLUCOSE_LO
+} from '../../../lib/drivers/abbott/freeStyleLibreConstants';
+
+const factoryConfigJson = `
+  {
+    "dataLength": 237,
+    "opCode": 81,
+    "packetLength": 240,
+    "data": {"type":"Buffer","data":[1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,70,114,101,101,83,116,121,108,101,32,76,105,98,114,101,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,4,10,3,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,0,0,1,1,1,2,0,0,246,1,38,0,60,0,60,0,0,0,240,0,70,0,0,60,1,0,128,213,224,80,0,0,60,0,0,0,0,0,87,23,0,0,100,11,0,0,125,0,203,0,224,46,0,10,45,1,0,0,128,175,27,0,180,0,30,0,10,6,5,10,35,35,35,40,35,40,35,50,50,35,35,35,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,79,130,30,27]}
+  }
+`;
+
+function deserialize(jsonString) {
+  return JSON.parse(jsonString, (k, v) => {
+    if (v !== null && typeof v === 'object' && 'type' in v &&
+      v.type === 'Buffer' && 'data' in v && Array.isArray(v.data)) {
+      // re-create Buffer objects for data fields of aapPackets
+      return new Buffer(v.data);
+    }
+    return v;
+  });
+}
+
+describe('freeStyleLibreData.js', () => {
+  const cfg = {
+    builder: builder(),
+    timezone: 'Europe/Berlin'
+  };
+
+  describe('non-static', () => {
+    let fsLibreData;
+
+    beforeEach(function(){
+      fsLibreData = new FreeStyleLibreData(cfg);
+    });
+
+    describe('process AAP packets', () => {
+
+      it('correctly restores 32bit records numbers', () => {
```

Comment:
eslint will probably warn that `it` scenarios should start with `should`. eg, "should correctly restore 32bit record numbers"